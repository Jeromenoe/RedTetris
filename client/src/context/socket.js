import React from 'react';
import socketio from "socket.io-client";


const localHost = "http://localhost:3001"; 
const herokuHost = "https://redtetris42-heroku.herokuapp.com"
const socketURL = process.env.NODE_ENV === "production" ? herokuHost : localHost

export const socket = socketio.connect(socketURL);
export const SocketContext = React.createContext();