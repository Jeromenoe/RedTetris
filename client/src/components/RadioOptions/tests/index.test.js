import React from 'react';
import { RadioOptions } from '../index';
import renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import {Provider} from 'react-redux';

import toJson from "enzyme-to-json";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount, configure } from 'enzyme';

import { SocketContext, socket } from "../../../context/socket"


configure({adapter: new Adapter()});


describe('Testing RadioOptions Component', () => {
	test('Test rendering ok', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				difficulty: 1,
				nbOfNextTetromino: 1,
				removeOwnIndestructibleLine: 1
			}
		};
		const store = mockStore(initialState);
		const wrapper = mount(<Provider store={store}><SocketContext.Provider value={socket}><RadioOptions settingId={1} /></SocketContext.Provider></Provider>);
		expect(toJson(wrapper)).toMatchSnapshot();
	  });

	  test('Test on game difficulty', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				difficulty: 1,
				nbOfNextTetromino: 1,
				removeOwnIndestructibleLine: 1
			}
		};
		const store = mockStore(initialState);
		
		const wrapper = mount(<Provider store={store}><SocketContext.Provider value={socket}><RadioOptions settingId={1} /></SocketContext.Provider></Provider>);
		wrapper.find('#opt-1').simulate('change');
		wrapper.find('#opt-2').simulate('change');
		wrapper.find('#opt-3').simulate('change');
		wrapper.find('#opt-4').simulate('change');
	});

	test('Test on nb of next tetrominos', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				difficulty: 1,
				nbOfNextTetromino: 1,
				removeOwnIndestructibleLine: 1
			}
		};
		const store = mockStore(initialState);
		
		const wrapper = mount(<Provider store={store}><SocketContext.Provider value={socket}><RadioOptions settingId={2} /></SocketContext.Provider></Provider>);
		wrapper.find('#opt-5').simulate('change');
		wrapper.find('#opt-6').simulate('change');
		wrapper.find('#opt-7').simulate('change');
		wrapper.find('#opt-8').simulate('change');
		wrapper.find('#opt-9').simulate('change');
	});

	test('Test on indestructible line option', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				difficulty: 1,
				nbOfNextTetromino: 1,
				removeOwnIndestructibleLine: 1
			}
		};
		const store = mockStore(initialState);
		
		const wrapper = mount(<Provider store={store}><SocketContext.Provider value={socket}><RadioOptions settingId={3} /></SocketContext.Provider></Provider>);
		wrapper.find('#opt-10').simulate('change');
		wrapper.find('#opt-11').simulate('change');
	});
});