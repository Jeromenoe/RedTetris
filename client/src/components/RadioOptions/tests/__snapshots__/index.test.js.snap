// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RadioOptions Component Test rendering ok 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <RadioOptions
    settingId={1}
  >
    <div
      className="radio-group"
    >
      <input
        checked={false}
        id="opt-1"
        name="selector-1"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="opt-1"
      >
        EASY
      </label>
      <input
        checked={true}
        id="opt-2"
        name="selector-1"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="opt-2"
      >
        MEDIUM
      </label>
      <input
        checked={false}
        id="opt-3"
        name="selector-1"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="opt-3"
      >
        HARD
      </label>
      <input
        checked={false}
        id="opt-4"
        name="selector-1"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="opt-4"
      >
        LEVEL
      </label>
    </div>
  </RadioOptions>
</Provider>
`;
